spec/system/candidate_interface/submitting/candidate_submitting_application_with_enic_interruption_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe 'Candidate submits the application with enic interruption pages' do
+  include CandidateHelper
+  include SignInHelper
+
+  scenario 'Candidate submits an application with all applications having an enic_reason of not_needed', :js do
+    given_i_am_signed_in
+
+    when_i_have_completed_my_application_and_have_added_primary_as_a_course_choice_with_not_needed_qualification
+    and_i_continue_with_my_application
+
+    when_i_save_as_draft
+    and_i_am_redirected_to_the_application_dashboard
+    and_my_application_is_still_unsubmitted
+    and_i_continue_with_my_application
+
+    when_i_click_to_review_my_application
+    then_i_see_a_interruption_page_for_personal_statement
+    when_i_continue_without_editing
+    then_i_see_a_interruption_page_for_not_needed_enic
+    then_the_viewed_enic_interruption_page_cookie_to_be_set
+  end
+
+  scenario 'Candidate submits an application with an application having an enic_reason of maybe', :js do
+    given_i_am_signed_in
+
+    when_i_have_completed_my_application_and_have_added_primary_as_a_course_choice_with_waiting_or_maybe_qualification
+    and_i_continue_with_my_application
+
+    when_i_save_as_draft
+    and_i_am_redirected_to_the_application_dashboard
+    and_my_application_is_still_unsubmitted
+    and_i_continue_with_my_application
+
+    when_i_click_to_review_my_application
+    then_i_see_a_interruption_page_for_personal_statement
+    when_i_continue_without_editing
+    then_i_see_a_interruption_page_for_waiting_or_maybe_enic
+    then_the_viewed_enic_interruption_page_cookie_to_be_set
+  end
```

Comment:
Can we add a test that include the scenario where the candidate sees all current interruption pages?

I need to add another for TDA soon and then we would have threee interruption pages.